portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -40,8 +41,25 @@
     ArrayList<Event> events = new ArrayList<>();
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Gson g = new Gson();
+    Key key = g.fromJson(request.getParameter("key"), com.google.appengine.api.datastore.Key.class);
+    key = KeyFactory.createKey("Event",key.getId()); 
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     Entity eventEntity = new Entity("Event");
```

Comment:
my preference would be to only create a new Entity when you know you need it, so like:

```
Entity eventEntity;
if (key == null) {
    eventEntity = new Entity("Event");
    key = eventEntity.getKey();
} 
else {
    try {
         eventEntity = datastore.get(key);
    }
...
}
```

It works fine as is but scares me a little bit to be creating a new Entity that we may accidentally store instead of the pre-existing one